cdap-api/src/main/java/co/cask/cdap/api/workflow/NodeValueEntry.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.workflow;
+
+/**
+ * Multiple nodes in the Workflow can add the same key to the {@link WorkflowToken}.
+ * This class provides a mapping from node name to the value which was set for the
+ * specific key.
+ */
+public final class NodeValueEntry {
+  private final String nodeName;
+  private final String value;
+
+  public NodeValueEntry(String nodeName, String value) {
+    this.nodeName = nodeName;
+    this.value = value;
+  }
+
+  public String getNodeName() {
+    return nodeName;
+  }
+
+  public String getValue() {
+    return value;
+  }
+
+  @Override
+  public String toString() {
+    StringBuilder sb = new StringBuilder("NodeValueEntry {");
+    sb.append("nodeName=").append(nodeName);
+    sb.append(", value=").append(value);
+    sb.append('}');
+    return sb.toString();
+  }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
Since the class `NodeValueEntry` itself only contains 2 members of type `String`, not sure how the `Objects.toString` would be better either. 

For example, current `toString` directly uses the `nodeName` and `value` instances. We should not use `Objects.toString(nodeName)` and `Objects.toString(value)` right? 

Had `nodeName` and `value` been the instances of another object, then yes you are right we would require `Objects.toString`.

Same with the `hashCode` method.

Please let me know if I am missing anything here.